### DISCUSSION TITLE
help with compiling firmware

### DISCUSSION THREAD

**Discussion by blackoutlaw** [Jyers/Marlin#1068] [Q&A] [answered]
ok i will start by saying that vscode has changed and now i am having trouble even getting the source code to pull up.  with that said i was trying to compile the lastest jayer firmware for a non ubl leveling with the bltouch for a bed of 400 x 400 x 450 on the 3v2 with a 4.2.2 board and i am having no luck trying to get it done.  i did get the source code to compile without changes but can't find where it put the bin file then when i try to compile the change of bed size it fails.  i am starting with the bltouch 3x3 non hs config files.  any help with this issue would be a great help as i would love to be able to compile my own when new updates come out since i have the extended bed size now.  thanks

**Comment by smooreace** [comment]
Every tutorial out there tells you the output file is
in \.pio\build\STM32F103RET6_creality. Not sure what else your having
trouble with because you didn't say what you did that had issues

**Comment by smooreace** [comment]
I will say that the easiest way to go is to support Jyers work by becoming
a Patreon supporter, and from that you get access to his hourly
auto-compile system. You add your config files and it pops out firmware
hourly

**Comment by blackoutlaw** [comment, marked as answer]
my first issue is selecting the right board i cant seam to find STM32F103RET6_creality if this is not right then which board then next is how do i load the source code to the project everything else i can do

**Reply by Cal-LF** [reply]
Unzip the source code and copy the correct config files to the Martin folder. Drag the folder onto VS Code window. Wait for it to settle. Change bed size in config and compile.

**Reply by smooreace** [reply]
Your running a 4.2.2 board, so there is nothing pertaining to that needing changed. All of Jyers pre-built configs are already set up for the 4.2.2 main-board
